### DISCUSSION TITLE
Can not import the wormhole wrapped coins in my smart contract

### DISCUSSION THREAD

**Discussion by seiji0411** [aptos-labs/aptos-developer-discussions#367] [Questions]
### Describe your question in detail.

https://aptoscan.com/account/28b98af2e2b8a6387ee392773135e0bf4b1f05e4c0c79ff51ab84a73e2454be8?module=stable_swap_adapter&viewMode=view#modules
I found that this module has the imported Wormhole wrapped coins.

```code
use WrappedCoin1::coin::T as WUSDC; // USD Coin (Wormhole)
use WrappedCoin2::coin::T as WWETH; // Wrapped Ether (Wormhole)
use WrappedCoin3::coin::T as WWTBT; // wTBT Pool
```

I tried to import Wrapped coins in my project as dependencies.
But I did not success. Having error `Move abort in 0x1::code: EPACKAGE_DEP_MISSING(0x60005): Dependency could not be resolved to any published package`

These are wrapped coin files
- Move.toml
```code
[package]
name = "WrappedCoin1"
version = "0.0.1"

[dependencies]
AptosFramework = { git = "https://github.com/aptos-labs/aptos-core.git", subdir = "aptos-move/framework/aptos-framework/", rev = "aptos-cli-v3.3.1" }

[addresses]
wrapped_coin_1="5e156f1207d0ebfa19a9eeff00d62a282278fb8719f4fab3a586a0a2c0fffbea"
```

- coin.move
```code
module wrapped_coin_1::coin {
    struct T has key {}
}
```

What is wrong?

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by Alexzander-Stone** [comment]
Hey there @seiji0411. Have you defined the dependencies in your project's `Move.toml` file? 

I recommend giving the docs a peek on [Third Party Dependencies](https://aptos.dev/en/build/smart-contracts/third-party-dependencies), as well as this [multidex-router example](https://github.com/aptos-labs/move-by-examples/tree/main/multidex-router). Both should help a ton with your use-case!

**Reply by seiji0411** [reply]
Thanks. @Alexzander-Stone 

I already defined the dependences in my project
```code
[dependencies]
WrappedCoin = { local = "../wrapped_coin1" }
``` 
About other modules, dependences are working fine.
But, it is the special case about this wrapped tokens.
Wormhole deployed this module in smart contract while running "create_wrapped_token_type" function.
I found that this module address on code is different with real on-chain module address.

I am wondering how Hippo team imported the wormhole wrapped tokens.

**Reply by seiji0411** [reply]
all wrapped token module code is the following:
```code
module 0x277fa055b6a73c42c0662d5236c65c864ccbf2d4abd21f174a30c8b786eab84b::coin {
    struct T has key {}
}
```

But, the deployed addresses are all different

WrappedCoin1::coin::T as WUSDC; // USD Coin (5e156f1207d0ebfa19a9eeff00d62a282278fb8719f4fab3a586a0a2c0fffbea)
WrappedCoin2::coin::T as WWETH; // Wrapped Ether (cc8a89c8dce9693d354449f1f73e60e14e347417854f029db5bc8e7454008abb)
WrappedCoin3::coin::T as WWTBT; // wTBT Pool (d916a950d4c1279df4aa0d6f32011842dc5c633a45c11ac5019232c159d115bb)
